File: javascript/codeeval/041_dup.js
 function d(str_arr) {
    var inputs = str_arr.split(';');
    var len = parseInt(inputs[0]);
    var arr = inputs[1].split(',');
    var res = {};
    for (var i = 0; i < len; i++) {
        if (res[arr[i]]) {
            console.log(arr[i]);
            break;
        } else {
            res[arr[i]] = 1;
        }
    }
}

var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        d(line);
    }
});

File: ruby/codeeval/041_dup.rb
def d(line)
    inputs = line.split(';')
    arr = inputs[1].strip().split(',')
    res = {}
    arr.each do |i|
        if res[i]
            puts(i)
            return
        else
            res[i] = 1
        end
    end
end


File.open(ARGV[0]).each_line do |line|
    d(line)
end
